cadc-util/src/main/java/ca/nrc/cadc/db/version/KeyValueDAO.java

```diff
@@ -197,6 +198,41 @@ public void delete(String name) {
         jdbc.update(sql, arg);
     }
 
+    public List<KeyValue> list() {
+        String select = String.format("SELECT %s, %s, %s FROM %s", columnNames[0], columnNames[1], columnNames[2], tableName);
+        ResultSetExtractor<List<KeyValue>> keyValueListExtractor = new KeyValueListExtractor();
+        List<KeyValue> keyValues;
+        try {
+            keyValues = jdbc.query(select, keyValueListExtractor);
+        } catch (Exception ex) {
+            Connection con = null;
+            try {
+                log.debug("query fail - check if table exists: " + tableName);
```

Comment:
Now that we are intending to use this for more than just the InitDatabase + ModelVersionDAO use case, this code to check for table existence and return null for various reasons) isn't that great anymore.

Better plan:
1. write a single method that checks for table existence and use that explicitly in InitDatabase; in all other cases table not existing is an error
2. allow other methods to throw spring-jdbc runtime exception but document that in the javadocs

Reply:
Updated the KeyValueDAO to allow the methods to throw the Spring exception on error, and check if the table exists in InitDatabase.